src/FINDOLOGIC/Helpers/ParameterBuilder.php

```diff
@@ -0,0 +1,449 @@
+<?php
+
+namespace FINDOLOGIC\Helpers;
+
+use FINDOLOGIC\Definitions\OrderType;
+use FINDOLOGIC\Definitions\RequestType;
+use FINDOLOGIC\Exceptions\InvalidParamException;
+use InvalidArgumentException;
+
+class ParameterBuilder
+{
+    const SHOPKEY = 'shopkey';
+
+    const HTTP_CLIENT = 'httpClient';
+
+    const API_URL = 'apiUrl';
+
+    const ALIVETEST_TIMEOUT = 'alivetestTimeout';
+
+    const REQUEST_TIMEOUT = 'requestTimeout';
+
+    protected $config = [
+        self::SHOPKEY,
+        self::API_URL => self::DEFAULT_API_URL,
+        self::ALIVETEST_TIMEOUT => self::DEFAULT_ALIVETEST_TIMEOUT,
+        self::REQUEST_TIMEOUT => self::DEFAULT_REQUEST_TIMEOUT,
+        self::HTTP_CLIENT
+    ];
+
+    const SHOP_URL = 'shopurl';
+
+    const USER_IP = 'userip';
+
+    const REFERER = 'referer';
+
+    const REVISION = 'revision';
+
+    const QUERY = 'query';
+
+    const ATTRIB = 'attrib';
+
+    const ORDER = 'order';
+
+    const PROPERTIES = 'properties';
+
+    const PUSH_ATTRIB = 'pushAttrib';
+
+    const COUNT = 'count';
+
+    const FIRST = 'first';
+
+    const IDENTIFIER = 'identifier';
+
+    const GROUP = 'group';
+
+    const FORCE_ORIGINAL_QUERY = 'forceOriginalQuery';
+
+    const INDIVIDUAL_PARAM = 'individualParam';
+
+    // Defaults
+    /** @var string URL Convention is https://API_URL/ps/SHOP_URL/ACTION.php */
+    const DEFAULT_API_URL = 'https://service.findologic.com/ps/%s/%s';
```

Comment:
I would say it is both, since it is set by me as default value if not explicitly overwritten in the config. But it is also a template, you're right. So it's a `DEFAULT_TEMPLATE_API_URL`.